packages/ckeditor5-dev-release-tools/lib/utils/executeinparallel.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const crypto = require( 'crypto' );
+const path = require( 'path' );
+const fs = require( 'fs' );
+const { Worker } = require( 'worker_threads' );
+const glob = require( 'glob' );
+const { tools } = require( '@ckeditor/ckeditor5-dev-utils' );
+
+const WORKER_SCRIPT = path.join( __dirname, 'parallelworker.js' );
+
+/**
+ * This util allows executing a specified task in parallel using Workers. It can be helpful when executing a not resource-consuming
+ * task in all packages specified in a path.
+ *
+ * If the callback loads dependencies, they must be specified directly in the function due to the worker's limitations.
+ * Functions cannot be passed to workers. Hence, we store the callback as a Node.js file loaded by workers.
+ *
+ * @see https://nodejs.org/api/worker_threads.html
+ * @param {Object} options
+ * @param {String} options.packagesDirectory
+ * @param {String} options.processDescription
+ * @param {Function} options.callback
+ * @param {AbortSignal} options.signal
+ * @param {String} [options.cwd=process.cwd()]
+ * @param {Number} [options.concurrency=require( 'os' ).cpus().length / 2]
+ * @returns {Promise}
+ */
+module.exports = function executeInParallel( options ) {
+	const {
+		packagesDirectory,
+		processDescription,
+		signal,
+		taskToExecute,
+		cwd = process.cwd(),
+		concurrency = require( 'os' ).cpus().length / 2
```

Comment:
What about `require( 'os' ).cpus().length - 1` to maximize speed?

Reply:
Would you like to measure if it speeds up significantly?

Reply:
It is dependent on the processor. When I made some tests with the number of cores, using 8 cores was about 20% faster than using 4 cores.
I just wrote it to say that when more cores are available, then why not to use them?
Leaving 1 cpu unused so that the CPU is responsive seams seasonable.

Reply:
Usually, we go through the release on a call. As we know, zoom takes some resources while sharing a screen. I doubt that it will be faster than when using a lower number of CPUs. But I got your point, and this option will be configurable via CLI, so an integrator can use it as much as possible.